### PR TITLE
Fix remaining MyPy errors in Google Provider

### PR DESCRIPTION
**Pull request by potiuk**
Part of #19891

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/compute.py [hunk 1]
```diff
@@ -405,10 +405,11 @@ def get_instance_address(
...
             return instance_info["networkInterfaces"][0].get("networkIP")
-
-        access_config = instance_info.get("networkInterfaces")[0].get("accessConfigs")
-        if access_config:
-            return access_config[0].get("natIP")
+        network_interfaces = instance_info.get("networkInterfaces")
+        if network_interfaces:
+            access_config = network_interfaces[0].get("accessConfigs")
+            if access_config:
+                return access_config[0].get("natIP")
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/compute.py]: Shame https://www.python.org/dev/peps/pep-0505/ doesn't exist yet.

File: airflow/providers/google/cloud/hooks/gcs.py [hunk 2]
```diff
@@ -114,6 +116,13 @@ def _inner_wrapper(self: "GCSHook", *args, **kwargs) -> RT:
     return _wrapper
 
 
+# A fake bucket to use in functions decorated by provide_session. This allows
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/gcs.py]: `_fallback_object_url_to_object_name_and_bucket_name` I think you mean, not provide_session :)

File: airflow/providers/google/cloud/hooks/os_login.py [hunk 3]
```diff
@@ -53,7 +54,13 @@ def get_conn(self) -> OsLoginServiceClient:
 
     @GoogleBaseHook.fallback_to_default_project_id
     def import_ssh_public_key(
-        self, user: str, ssh_public_key: Dict, project_id: str, retry=None, timeout=None, metadata=None
+        self,
+        user: str,
+        ssh_public_key: Dict,
+        project_id: str,
```
Comment (ashb) [inline airflow/providers/google/cloud/hooks/os_login.py]: Should `project_id` have the same default-cast-None trick as bucket?

**Review by ashb — APPROVED** [review]
Couple of small comments, but LGTM once those are dealt with

File: airflow/providers/google/cloud/hooks/compute.py (hunk 1 above)
Comment (uranusjr) [reply airflow/providers/google/cloud/hooks/compute.py]: It’s not likely to ever exist

Comment (ashb) [reply airflow/providers/google/cloud/hooks/compute.py]: Yeah fair.

Comment (potiuk) [reply airflow/providers/google/cloud/hooks/compute.py]: Why? I really would like to have some of those concepts in PEP 505 implemented. Are there good reasons it's not?

File: airflow/providers/google/cloud/hooks/os_login.py (hunk 3 above)
Comment (potiuk) [reply airflow/providers/google/cloud/hooks/os_login.py]: good point - looks like no tests of ours have missing project_id :)

**Pull request merged**